### PR TITLE
Enabling Azure DevOps YAML builds.

### PR DESCRIPTION
**Pull request by livarcocc**

### CODE REVIEW

File: .vsts-ci.yml [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+variables:
+  teamName: Roslyn-Project-System
+
+phases:
+- template: /eng/build.yml
+  parameters:
+    agentOs: Windows_NT
+    queue:
+      name: Hosted VS2017
+      parallel: 2
```
Comment (chcosta) [inline .vsts-ci.yml]: you have 4 legs of the matrix, do you want to change this to 4 or even to 99 so you don't have to fiddle with it and these always run in parallel if there are machines available?

File: .vsts-ci.yml [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+          _BuildConfig: Release
+          _BuildArchitecture: x64
+          _DOTNET_CLI_UI_LANGUAGE: ''
+          _DropSuffix: true
+
+- template: /eng/build.yml
+  parameters:
+    agentOs: Linux
+    queue:
+      name: Hosted Ubuntu 1604
```
Comment (chcosta) [inline .vsts-ci.yml]: add `parallel: 99`

File: eng/build.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  agentOs: ''
+  queue: {}
+  variables: {}
+
+phases:
+- template: /eng/common/templates/phases/base.yml
+  parameters:
+    name: ${{ parameters.agentOs }}
+    enableTelemetry: true
+    enableMicrobuild: true
```
Comment (chcosta) [inline eng/build.yml]: this looks like it's for public builds only (ie, I don't see any reference to an internal windows build queue).  Is this intended to be just for public builds (if so, then this should just be omitted)

File: eng/build.yml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    publicBuildReasons: PullRequest
+    queue: ${{ parameters.queue }}
+    variables: 
+      ${{ insert }}: ${{ parameters.variables }}
+      _AgentOSName: ${{ parameters.agentOs }}
+      _HelixType: build/product
+      _HelixBuildConfig: $(_BuildConfig)
+      BuildConfig: $(_BuildConfig)
+      BuildArchitecture: $(_BuildArchitecture)
+      _HelixSource: pr/dotnet/toolset/$(Build.SourceBranch)
```
Comment (chcosta) [inline eng/build.yml]: I'd expect helixsource to be different for public vs internal builds

File: eng/common/templates/phases/base.yml [hunk 5]
```diff
@@ -0,0 +1,107 @@
+parameters:
```
Comment (chcosta) [inline eng/common/templates/phases/base.yml]: I presume all the stuff under `eng/common` is just a straight port from Arcade, I don't intend to review it.

**Comment by livarcocc** [comment]
This PR is just for public CI. We will update it later for internal build, but we are not going to turn on internal builds now, so I preferred to have the PR focus on its goal.

File: eng/common/templates/phases/base.yml (hunk 5 above)
Comment (livarcocc) [reply eng/common/templates/phases/base.yml]: Correct.

File: eng/build.yml (hunk 4 above)
Comment (livarcocc) [reply eng/build.yml]: it will be, once we add internal builds. This file only covers public builds.

File: eng/build.yml (hunk 3 above)
Comment (livarcocc) [reply eng/build.yml]: Will remove it.

File: .vsts-ci.yml (hunk 2 above)
Comment (livarcocc) [reply .vsts-ci.yml]: Done.

**Pull request merged**